### PR TITLE
Meat Hook Rework (Accidental Features)

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Or, how I tried to kill `/datum/forced_movement` but got absolutely clonged.

Actually, I did kill `/datum/forced_movement`. It was only used in one spot so I just went ahead and cooked it into a special utility datum that should only be used in one spot. We can optimize the code later or something, but I like the way it is right now (pretty much status quo without the potential of someone using a depreciated framework).

Alright, there were also like 3 `TODO`s (4 if you count the move loop comment (which is ehhhh)). I naively tried to tackle them a very hard way, but then I just realized I could use the fancy new datum I cooked up and wow they're all solved now. The hook looks so fucking good now.

* The code is overall more streamlined, with all of the post-projectile work being handled by a special datum (I wanted it to be handled by the hook but the timings were all based on SSFastprocess so datum it is). Forced movement is killed but we just salvaged whatever we needed from it.
* More traits to ensure we don't cheese in a way we're not supposed to
* A very sexy chain will spawn when you drag your kill in (this wasn't there before but I fixeded it :3)
* The firer will actually get grounded when they try and shoot the chain out. They get grounded for 0.25 seconds unless they hit something. I don't know how the timing is but I think it's fair.
* We also add a unique suicide act, because I noticed we did the "magical" one previously, which big-league sucked.
* More traits to ensure less cheese! I like how nice it is now.
## Why It's Good For The Game

The meat hook really makes you _feel_ like Roadhog from Overwatch. Resolves a bunch of old TODOs while getting rid of a completely obsolete framework in a really neat way. I don't typically like mixing balances and refactors but these TODOs were getting crusty man i just wanted to get them out of the way
## Changelog
:cl:
balance: The Meat Hook will now "ground" you whenever you fire it out at someone. You get a very small immobilization every time you fire, which "upgrades" to a full immobilization whenever you actually hit an atom and start to drag it in.
fix: A chain should now show up as you drag in something with the meat hooks.
fix: Meat hooks should no longer play the "magical gun" suicide if you were to use it, but instead do their own unique thing.
/:cl:

### CODE REVIEW

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -45,7 +45,6 @@
 	var/atom/movable/moving_from_pull
 	///Holds information about any movement loops currently running/waiting to run on the movable. Lazy, will be null if nothing's going on
 	var/datum/movement_packet/move_packet
-	var/datum/forced_movement/force_moving = null //handled soley by forced_movement.dm
```
Comment (san7890) [inline code/game/atoms_movable.dm:48]: This was the real reason I made this PR to begin with - reducing the amount of crud we have to scroll through on view variables stuff

**Comment by AdipemDragon** [comment]
And here I thought until halfway through the pr that you were talking about the meat hooks you butcher shit on for meat 😭

**Comment by DaCoolBoss** [comment]
Visible chain when reeling someone in was sorely needed. Big improvement to the Pudge/Blitzcrank meta.

**Comment by Autisem** [comment]
>roadhog from overwatch

If only he knew

**Pull request merged**